### ISSUE TITLE
Passing an async lamba and passing a reference to its Func<Task> do not behave the same

### ISSUE THREAD

**Issue by heathbm** [dotnet/roslyn-analyzers#5981]
## Issue 1: Code appears correct, however actual outcome can be unexpected:

An async lambda `async () => ...` will resolve to be a `Func<Task>`. 
When this is provided to a method that only expects an `Action`, the method will treat it as an `Action` with no visible warnings. This can cause unexpected behavior from the perspective of a developer / API consumer.

The following example below will fail because `Task.cs` does not have a constructor with a `Func<Task>` parameter, only `Action`: https://github.com/microsoft/referencesource/blob/master/mscorlib/system/threading/Tasks/Task.cs
The section `Task.Delay(oneSecond)` is not awaited because the `Func<Task>` is treated as an `Action`.

Example:

```C#
[Fact]
async Task TaskStartFails()
{
    int taskProgress = 0;
    var oneSecond = TimeSpan.FromSeconds(1);

    var aTask = new Task(async () =>
    {
        await Task.Delay(oneSecond);
        Interlocked.Increment(ref taskProgress);
    });

    aTask.Start();
    await aTask;

    Assert.Equal(1, taskProgress); // Fail: taskProgress will be 0
}
```

## Issue 2: Inconsistency:

The following example does not compile due to an error.
Providing an async lambda directly and providing a reference to it do not behave consistently as one would expect.

```C#
using System;
using System.Threading.Tasks;
	
public class Program
{
    public static void Main()
    {
        OtherMethod(async () => await Task.Delay(100)); // No errors as it is passed directly and treated as an Action.

        var anAsyncAnonFunc = async () => await Task.Delay(100);
        OtherMethod(anAsyncAnonFunc); // Error: CS1503: Argument 1: cannot convert from 'System.Func<System.Threading.Tasks.Task>' to 'System.Action'
    }

    static void OtherMethod(Action anAction)
    {
        anAction();
    }
}
```

### Describe suggestions on how to achieve the rule

I believe a warning would be appropriate, along the lines of `Func<Task> is being treated as an Action and will not be awaited.`;

### Additional context

None.

**Comment by mavasani** [comment]
This needs to be triaged by dotnet/runtime. Transferring the issue.
